module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,14.93,True,2023-05-31T07:00:21,,
cowidev.vax.batch.australia,,,2023-05-31T07:00:06,,
cowidev.vax.batch.austria_old,,,2023-05-31T07:00:06,,
cowidev.vax.batch.belgium,10.21,True,2023-05-31T07:00:17,,
cowidev.vax.batch.bolivia,4.26,True,2023-05-31T07:00:11,,
cowidev.vax.batch.canada,7.02,False,2023-05-31T07:02:30,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 282, in main
    Canada().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 270, in export
    df = df.pipe(self.pipeline, df_man)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 250, in pipeline
    .pipe(self.pipe_make_monotonic)[
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 230, in pipe_make_monotonic
    df = df.pipe(self.make_monotonic, max_removed_rows=self.max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
",1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
cowidev.vax.batch.chile,0.96,True,2023-05-31T07:00:07,,
cowidev.vax.batch.czechia,38.29,True,2023-05-31T07:00:45,,
cowidev.vax.batch.denmark,,,2023-05-31T07:00:06,,
cowidev.vax.batch.ecdc,94.34,True,2023-05-31T07:01:41,,
cowidev.vax.batch.ecuador,,,2023-05-31T07:00:06,,
cowidev.vax.batch.estonia,0.6,True,2023-05-31T07:00:07,,
cowidev.vax.batch.france,9.15,True,2023-05-31T07:00:15,,
cowidev.vax.batch.germany,0.16,True,2023-05-31T07:00:06,,
cowidev.vax.batch.greece,15.0,True,2023-05-31T07:00:21,,
cowidev.vax.batch.hong_kong,18.07,True,2023-05-31T07:00:24,,
cowidev.vax.batch.indonesia,,,2023-05-31T07:00:06,,
cowidev.vax.batch.ireland,4.7,True,2023-05-31T07:00:11,,
cowidev.vax.batch.israel,10.21,True,2023-05-31T07:00:17,,
cowidev.vax.batch.italy,10.17,True,2023-05-31T07:00:16,,
cowidev.vax.batch.japan,10.18,False,2023-05-31T07:02:40,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 375, in main
    Japan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 359, in export
    df = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 71, in read
    df_latest = self.read_latest().pipe(self.pipe_read_latest)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 100, in read_latest
    self._read_latest_booster(),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 211, in _read_latest_booster
    df = self._read_xlsx(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 233, in _read_xlsx
    raise ValueError(f""Unknown sheets: {sheets_unknown}"")
ValueError: Unknown sheets: {'６回目_一般接種回数', '５回目_総接種回数', '６回目_総接種回数', '5.8以降接種_総接種回数', '5.8以降接種_一般接種（４回目）', '5.8以降接種_一般接種（５回目）', '５回目_一般接種回数', '5.8以降接種_一般接種（６回目）', '5.8以降接種_一般接種（３回目）'}
","Unknown sheets: {'６回目_一般接種回数', '５回目_総接種回数', '６回目_総接種回数', '5.8以降接種_総接種回数', '5.8以降接種_一般接種（４回目）', '5.8以降接種_一般接種（５回目）', '５回目_一般接種回数', '5.8以降接種_一般接種（６回目）', '5.8以降接種_一般接種（３回目）'}"
cowidev.vax.batch.jersey,7.67,True,2023-05-31T07:00:14,,
cowidev.vax.batch.latvia,68.92,True,2023-05-31T07:01:15,,
cowidev.vax.batch.lithuania,5.95,True,2023-05-31T07:00:12,,
cowidev.vax.batch.luxembourg,3.35,True,2023-05-31T07:00:10,,
cowidev.vax.batch.malaysia,0.8,True,2023-05-31T07:00:07,,
cowidev.vax.batch.malta,1.24,True,2023-05-31T07:00:08,,
cowidev.vax.batch.netherlands,,,2023-05-31T07:00:06,,
cowidev.vax.batch.new_zealand,16.21,True,2023-05-31T07:00:23,,
cowidev.vax.batch.norway,0.49,True,2023-05-31T07:00:07,,
cowidev.vax.batch.peru,4.92,True,2023-05-31T07:00:11,,
cowidev.vax.batch.portugal,,,2023-05-31T07:00:06,,
cowidev.vax.batch.romania,37.98,True,2023-05-31T07:00:44,,
cowidev.vax.batch.saudi_arabia,10.07,True,2023-05-31T07:00:16,,
cowidev.vax.batch.singapore,5.83,True,2023-05-31T07:00:12,,
cowidev.vax.batch.slovakia,50.31,True,2023-05-31T07:00:57,,
cowidev.vax.batch.slovenia,3.83,True,2023-05-31T07:00:10,,
cowidev.vax.batch.south_korea,56.47,False,2023-05-31T07:03:37,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 209, in main
    SouthKorea().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 191, in export
    df_base = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 43, in read
    dfs = pd.read_html(to_proxy_url(self.source_url), encoding=""utf-8"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/util/_decorators.py"", line 311, in wrapper
    return func(*args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/io/html.py"", line 1098, in read_html
    return _parse(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/io/html.py"", line 906, in _parse
    tables = p.parse_tables()
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/io/html.py"", line 222, in parse_tables
    tables = self._parse_tables(self._build_doc(), self.match, self.attrs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/io/html.py"", line 745, in _build_doc
    raise e
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/io/html.py"", line 726, in _build_doc
    with urlopen(self.io) as f:
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/io/common.py"", line 212, in urlopen
    return urllib.request.urlopen(*args, **kwargs)
  File ""/usr/lib/python3.10/urllib/request.py"", line 216, in urlopen
    return opener.open(url, data, timeout)
  File ""/usr/lib/python3.10/urllib/request.py"", line 525, in open
    response = meth(req, response)
  File ""/usr/lib/python3.10/urllib/request.py"", line 634, in http_response
    response = self.parent.error(
  File ""/usr/lib/python3.10/urllib/request.py"", line 563, in error
    return self._call_chain(*args)
  File ""/usr/lib/python3.10/urllib/request.py"", line 496, in _call_chain
    result = func(*args)
  File ""/usr/lib/python3.10/urllib/request.py"", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 500: Internal Server Error
",HTTP Error 500: Internal Server Error
cowidev.vax.batch.spc,39.53,True,2023-05-31T07:00:46,,
cowidev.vax.batch.sweden,12.45,True,2023-05-31T07:00:19,,
cowidev.vax.batch.switzerland,136.39,True,2023-05-31T07:02:23,,
cowidev.vax.batch.trinidad_and_tobago,4.97,True,2023-05-31T07:00:12,,
cowidev.vax.batch.ukraine,,,2023-05-31T07:00:07,,
cowidev.vax.batch.united_kingdom,16.97,True,2023-05-31T07:00:24,,
cowidev.vax.batch.united_states,11.64,True,2023-05-31T07:00:19,,
cowidev.vax.batch.uruguay,6.94,True,2023-05-31T07:00:17,,
cowidev.vax.batch.zimbabwe,3.9,True,2023-05-31T07:00:14,,
cowidev.vax.incremental.africacdc,,,2023-05-31T07:00:11,,
cowidev.vax.incremental.antigua_barbuda,,,2023-05-31T07:00:11,,
cowidev.vax.incremental.aruba,1.68,True,2023-05-31T07:00:12,,
cowidev.vax.incremental.austria,,,2023-05-31T07:00:11,,
cowidev.vax.incremental.azerbaijan,,,2023-05-31T07:00:11,,
cowidev.vax.incremental.bahrain,,,2023-05-31T07:00:11,,
cowidev.vax.incremental.bangladesh,47.14,True,2023-05-31T07:00:58,,
cowidev.vax.incremental.barbados,,,2023-05-31T07:00:11,,
cowidev.vax.incremental.brazil,1.34,True,2023-05-31T07:00:13,,
cowidev.vax.incremental.bulgaria,7.08,True,2023-05-31T07:00:19,,
cowidev.vax.incremental.china,20.3,True,2023-05-31T07:00:32,,
cowidev.vax.incremental.costa_rica,,,2023-05-31T07:00:12,,
cowidev.vax.incremental.croatia,2.33,True,2023-05-31T07:00:15,,
cowidev.vax.incremental.cuba,26.09,True,2023-05-31T07:00:38,,
cowidev.vax.incremental.curacao,,,2023-05-31T07:00:13,,
cowidev.vax.incremental.cyprus,3.22,True,2023-05-31T07:00:16,,
cowidev.vax.incremental.dominican_republic,13.03,False,2023-05-31T07:03:50,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 112, in main
    DominicanRepublic().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 26, in read
    metrics = self._parse_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 35, in _parse_metrics
    metric_candidates = _find_potential_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 91, in _find_potential_metrics
    h3 = _find_h3(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 81, in _find_h3
    raise ValueError(""More than one element found!"")
ValueError: More than one element found!
",More than one element found!
cowidev.vax.incremental.el_salvador,6.5,True,2023-05-31T07:00:21,,
cowidev.vax.incremental.faeroe_islands,1.26,True,2023-05-31T07:00:16,,
cowidev.vax.incremental.fiji,0.0,True,2023-05-31T07:00:15,,
cowidev.vax.incremental.finland,0.42,True,2023-05-31T07:00:16,,
cowidev.vax.incremental.gabon,,,2023-05-31T07:00:16,,
cowidev.vax.incremental.georgia,3.04,True,2023-05-31T07:00:19,,
cowidev.vax.incremental.greenland,,,2023-05-31T07:00:16,,
cowidev.vax.incremental.guatemala,,,2023-05-31T07:00:16,,
cowidev.vax.incremental.guernsey,,,2023-05-31T07:00:16,,
cowidev.vax.incremental.hungary,82.83,True,2023-05-31T07:01:39,,
cowidev.vax.incremental.iceland,1.97,True,2023-05-31T07:00:18,,
cowidev.vax.incremental.india,1.67,True,2023-05-31T07:00:18,,
cowidev.vax.incremental.iran,,,2023-05-31T07:00:16,,
cowidev.vax.incremental.isle_of_man,2.66,True,2023-05-31T07:00:19,,
cowidev.vax.incremental.jamaica,10.67,True,2023-05-31T07:00:27,,
cowidev.vax.incremental.kazakhstan,10.06,True,2023-05-31T07:00:27,,
cowidev.vax.incremental.kosovo,,,2023-05-31T07:00:17,,
cowidev.vax.incremental.kyrgyzstan,2.48,True,2023-05-31T07:00:19,,
cowidev.vax.incremental.laos,1.68,True,2023-05-31T07:03:52,,
cowidev.vax.incremental.macao,6.84,False,2023-05-31T07:03:59,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 90, in main
    Macao().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 75, in export
    data = self.read()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 23, in read
    iframe_url = self._get_iframe_url(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 31, in _get_iframe_url
    elem = driver.find_element_by_id(""ICovid19Monitor"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 360, in find_element_by_id
    return self.find_element(by=By.ID, value=id_)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 321, in execute
    self.error_handler.check_response(response)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py"", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)

","Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)
"
cowidev.vax.incremental.moldova,,,2023-05-31T07:00:19,,
cowidev.vax.incremental.monaco,90.87,True,2023-05-31T07:01:50,,
cowidev.vax.incremental.mongolia,1.18,True,2023-05-31T07:00:20,,
cowidev.vax.incremental.montenegro,1.05,True,2023-05-31T07:00:20,,
cowidev.vax.incremental.morocco,,,2023-05-31T07:00:19,,
cowidev.vax.incremental.myanmar,,,2023-05-31T07:00:19,,
cowidev.vax.incremental.nepal,,,2023-05-31T07:00:19,,
cowidev.vax.incremental.north_macedonia,,,2023-05-31T07:00:19,,
cowidev.vax.incremental.northern_cyprus,7.51,True,2023-05-31T07:00:27,,
cowidev.vax.incremental.paho,83.15,False,2023-05-31T07:05:22,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 195, in main
    PAHO().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 190, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 167, in pipeline
    df.pipe(self.pipe_check_columns)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 122, in pipe_check_columns
    raise ValueError(f""Missing column fields: {columns_missing}. Present columns are: {df.columns}"")
ValueError: Missing column fields: {'1st additional dose', 'Second dose', 'Country/ Territory', 'Single dose', 'First dose', 'Total doses', 'Country code', '2nd additional dose'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')
","Missing column fields: {'1st additional dose', 'Second dose', 'Country/ Territory', 'Single dose', 'First dose', 'Total doses', 'Country code', '2nd additional dose'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')"
cowidev.vax.incremental.pakistan,,,2023-05-31T07:00:19,,
cowidev.vax.incremental.philippines,,,2023-05-31T07:00:19,,
cowidev.vax.incremental.poland,2.04,True,2023-05-31T07:00:21,,
cowidev.vax.incremental.qatar,,,2023-05-31T07:00:19,,
cowidev.vax.incremental.russia,28.64,True,2023-05-31T07:00:48,,
cowidev.vax.incremental.saint_lucia,,,2023-05-31T07:00:20,,
cowidev.vax.incremental.serbia,9.42,True,2023-05-31T07:00:30,,
cowidev.vax.incremental.south_africa,,,2023-05-31T07:00:20,,
cowidev.vax.incremental.spain,8.6,True,2023-05-31T07:00:29,,
cowidev.vax.incremental.sri_lanka,,,2023-05-31T07:00:21,,
cowidev.vax.incremental.suriname,15.74,True,2023-05-31T07:00:36,,
cowidev.vax.incremental.taiwan,5.51,False,2023-05-31T07:05:27,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 184, in main
    Taiwan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 170, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 38, in read
    data = self.parse_data(df, soup)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 122, in parse_data
    ""date"": self._parse_date(soup),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 156, in _parse_date
    date_str = re.search(regex, date_raw).group(1)
AttributeError: 'NoneType' object has no attribute 'group'
",'NoneType' object has no attribute 'group'
cowidev.vax.incremental.thailand,6.66,True,2023-05-31T07:00:28,,
cowidev.vax.incremental.turkey,,,2023-05-31T07:00:21,,
cowidev.vax.incremental.united_arab_emirates,,,2023-05-31T07:00:21,,
cowidev.vax.incremental.vietnam,,,2023-05-31T07:00:21,,
cowidev.vax.incremental.who,35.17,True,2023-05-31T07:00:56,,
cowidev.vax.incremental.zambia,2.77,True,2023-05-31T07:00:25,,
